controlpanel/api/elasticsearch.py

```diff
@@ -59,7 +68,13 @@ def app_logs(app, num_hours=None):
 
     s = s.params(preserve_order=True)
 
-    for entry in s.scan():
-        entry['timestamp'] = parse(entry['@timestamp'])
-        yield entry
 
+    try:
+        logs = list(s.scan())
+        log.info("Got {} log entries.".format(len(logs)))
+        for entry in logs:
+            entry['timestamp'] = parse(entry['@timestamp'])
+    except Exception as ex:
+        log.error(ex)
+        logs = []
```

Comment:
I wonder if we could add a message in this case, something like "Failed to get logs for this webapp" or if it's a bad idea?

Probably the wrong place add this logic, alternatively we let the exception bubble up here and handle in the view but I don't know if that makes things more complicated.

Reply:
The log message will include all the information to help us identify that this was to do with getting logs from Elastic (the module generating the exception is called something like `elastic_dsl` thus making it obvious what's going on).